src/io.c

```diff
@@ -137,6 +137,48 @@ static int pids[MAXCHLDS];         /* and this number! */
 static int fistats = 0;            /* First used entry */
 static int lastats = 0;            /* First unused entry */
 static int statsfull = 0;          /* Flag, whether stats FIFO full */
+
+static int findSignaledPid(int pidc)
+{
+    int pos = fistats;
```

Comment:
This function fails to check for an empty queue, which may lead to false positives if it instead treats the queue as full, as we do not reset "used up" entries before removing them from the queue.

I.e. just add
```C
// check if queue is empty
if (fistats == lastats && !statsfull)
  return -1;
```